test/e2e/framework/kcp.go

```diff
@@ -94,27 +94,42 @@ type kcpFixture struct {
 
 // PrivateKcpServer returns a new kcp server fixture managing a new
 // server process that is not intended to be shared between tests.
-func PrivateKcpServer(t *testing.T, options ...KcpConfigOption) RunningServer {
+func PrivateKcpServer(t *testing.T, opts ...PrivateKcpServerOption) RunningServer {
 	t.Helper()
-
 	serverName := "main"
 
 	cfg := &kcpConfig{Name: serverName}
-	for _, opt := range options {
-		cfg = opt(cfg)
-	}
-
-	if len(cfg.ArtifactDir) == 0 || len(cfg.DataDir) == 0 {
-		artifactDir, dataDir, err := ScratchDirs(t)
-		require.NoError(t, err, "failed to create scratch dirs: %v", err)
-		cfg.ArtifactDir = artifactDir
-		cfg.DataDir = dataDir
+	for _, opt := range opts {
+		opt(cfg)
 	}
+	cfg.Args = append(cfg.Args, TestServerWithAuditPolicyFile(WriteEmbedFile(t, "audit-policy.yaml"))...,)
 
 	f := newKcpFixture(t, *cfg)
 	return f.Servers[serverName]
 }
 
+type PrivateKcpServerOption func(c *kcpConfig)
```

Comment:
nit: maybe look at the type-union approach I used in `test/e2e/framework/workspaces.go` to discriminate between option types a little harder?

Reply:
To be honest in this case I'd rather go the opposite direction - unlike in your workspaces code (where there is a logical set of Privileged and Unprivileged options) here we have `kcpConfig` options which can potentially work regardless of whether the server instance is Private or Shared (the options are the same, it's just the usage of the resulting server that is different).

As a specific example, look at `PrivateWithScratchDirectories` vs the existing `WithScratchDirectories` - we don't really need both IMO, thus further discriminating between Private vs Shared probably doesn't make much sense?

This is carried over from work @ncdc started so I left it as-is for now, but I can either look at reworking this, or take an action to do it as a follow-up.

Reply:
I'm fine with either fixing it now or in a follow up

Reply:
Since it's not strictly related to the rest of this PR (and provides a way to clean up `TestAuditLogs`) I broke this part into https://github.com/kcp-dev/kcp/pull/2610

That removes `PrivateKcpServerOption` and just makes the existing `WithScratchDirectories` work with the PrivateKcpServer fixture.

Reply:
`PrivateKcpServerOption` now removed